### PR TITLE
poac: use legacysupport.use_mp_libcxx on Mojave and older

### PR DESCRIPTION
**Pull request by aeiouaeiouaeiouaeiouaeiouaeiou**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.8.5 12F2560 x86_64
Xcode 5.1.1 5B1008

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
Finally it build successfully on my old system, but stuck with this linking issue when starts:
```
➜  ~ poac --version
dyld: Library not loaded: /opt/local/lib/libpoac_cmd.dylib
  Referenced from: /opt/local/bin/poac
  Reason: image not found
[1]    98801 trace trap  poac --version
➜  ~ otool -L $(which poac)
/opt/local/bin/poac:
	/opt/local/lib/libpoac_cmd.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/local/lib/libgit2.1.3.dylib (compatibility version 1.3.0, current version 1.3.0)
	/opt/local/lib/libpoac_core.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/local/lib/libpoac_core_builder_ninja.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/local/libexec/boost/1.78/lib/libboost_regex-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/local/lib/libpoac_core_builder_compiler_cxx.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/local/lib/libpoac_core_builder_compiler_lang.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/local/lib/libpoac_data.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/local/lib/libpoac_core_resolver.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/local/lib/libarchive.13.dylib (compatibility version 20.0.0, current version 20.1.0)
	/opt/local/lib/libpoac_util_semver.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/local/libexec/openssl3/lib/libssl.3.dylib (compatibility version 3.0.0, current version 3.0.0)
	/opt/local/libexec/openssl3/lib/libcrypto.3.dylib (compatibility version 3.0.0, current version 3.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 169.3.0)
	/opt/local/lib/libfmt.8.dylib (compatibility version 8.0.0, current version 8.1.1)
	/opt/local/lib/libcxx/libc++.1.0.dylib (compatibility version 1.0.0, current version 1.0.0)
```
The `libpoac*` libraries are not installed for some reason. Do you have any idea @ken-matsui?

**Comment by ken-matsui** [comment]
@aeiouaeiouaeiouaeiouaeiouaeiou 

Thank you so much for working on it! I'll have a look into the error.

**Comment by ken-matsui** [comment]
@aeiouaeiouaeiouaeiouaeiouaeiou

I believe this will be fixed after https://github.com/macports/macports-ports/pull/15447 is merged and rebasing this PR.

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
Now it will be very strange if it does not work.

**Comment by mascguy** [comment]
> Now it will be very strange if it does not work.

While CI succeeded for Big Sur and Monterey - and in 25-ish minutes for each - the 10.15 job didn't seem too happy. Cancelled the later, and re-queued another run.

Let's see if it's any better this time around.

File: sysutils/poac/Portfile [hunk 1]
```diff
@@ -22,6 +27,16 @@ checksums           rmd160  13108a1b06d9c9cf7d4da27d49b121d255467d62 \
...
+# Do not allow CMake to use system Git on older systems
+if {${os.platform} eq "darwin" && ${os.major} < 18} {
+     set cmake_git       port:git
+} else {
+     set cmake_git       bin:git:git
+}
+
+depends_fetch-append \
+                    ${cmake_git}
+
```
Comment (mascguy) [inline sysutils/poac/Portfile:39]: If a port needs `git`, it's preferable to use the version packaged with MacPorts. So I'd recommend doing that across-the-board.

Comment (mascguy) [reply sysutils/poac/Portfile:33]: You don't want to use a `bin`-style dependency, as that will use the system git if installed. Instead, explicitly use `port:git`.

**Review by mascguy — APPROVED** [review]

**Comment by mascguy** [comment]
> While CI succeeded for Big Sur and Monterey - and in 25-ish minutes for each - the 10.15 job didn't seem too happy. Cancelled the later, and re-queued another run.

10.15 completed in 25 minutes this time, so that's encouraging!

Hopefully the Big Sur and Monterey builds will go smoothly too, since they did so previously.

**Comment by mascguy** [comment]
> 10.15 completed in 25 minutes this time, so that's encouraging!
> 
> Hopefully the Big Sur and Monterey builds will go smoothly too, since they did so previously.

Ugh, my eyes are failing me: The 10.15 job is running for an extended period of time once again.

Hmmm...

**Comment by ken-matsui** [comment]
MacPorts is trying to build LLVM... It would be better if we could use a binary or something like it.

**Comment by mascguy** [comment]
> MacPorts is trying to build LLVM... It would be better if we could use a binary or something like it.

That should be occurring already. I'll try to test these changes locally on 10.15 - sometime today, hopefully - to see what's happening.

**Comment by mascguy** [comment]
> > MacPorts is trying to build LLVM... It would be better if we could use a binary or something like it.
> 
> That should be occurring already. I'll try to test these changes locally on 10.15 - sometime today, hopefully - to see what's happening.

Builds fine locally on 10.15, so I'm not concern about that CI job. Also tested with trace mode enabled, and everything looks good.

**Comment by ken-matsui** [comment]
That's good to know. Thank you all!

**Pull request merged**